tensorflow/python/eager/function.py

```diff
@@ -2641,44 +2634,44 @@ def canonicalize_function_inputs(self, *args, **kwargs):
 
     if not kwargs:
       inputs = args
-      if self._arg_indices_to_default_values:
-        try:
-          inputs += tuple(
-              self._arg_indices_to_default_values[i]
-              for i in range(len(args), len(self._arg_names)))
-        except KeyError:
-          missing_args = [
-              self._arg_names[i]
-              for i in range(len(args), len(self._arg_names))
-              if i not in self._arg_indices_to_default_values
-          ]
+      if self._fullargspec.defaults:
+        if len(args) < self._num_req_args:
+          missing_args = self._arg_names[len(args):self._num_req_args]
           raise TypeError("{} missing required arguments: {}".format(
               self.signature_summary(), ", ".join(missing_args)))
+        inputs += tuple(
+            self._fullargspec.defaults[len(args) - self._num_req_args:])
 
       if self._fullargspec.kwonlydefaults:
         kwargs.update(self._fullargspec.kwonlydefaults)
     else:
-      # Maps from index of arg to its corresponding value, according to `args`
-      # and `kwargs`; seeded with the default values for the named args that
-      # aren't in `args`.
-      arg_indices_to_values = {
-          index: default for index, default in six.iteritems(
-              self._arg_indices_to_default_values) if index >= len(args)
-      }
-      consumed_args = []
+      add_args = [None] * (self._num_tot_args - len(args))
```

Comment:
Good variable naming is _very_ important for readable, maintainable code. `add_args` is an action and doesn't tell much about what it is.  It can be, for example, `remaining_kargs`

Reply:
Changed to `remaining_args`